admin-dev/themes/new-theme/js/pages/product/combination/bulk-form-handler.ts

```diff
@@ -0,0 +1,224 @@
+    this.init();
+  }
+
+  private init(): void {
+    this.listenCheckboxesChange();
+    this.eventEmitter.on(CombinationEvents.listRendered, () => this.toggleBulkActions());
+
+    const bulkFormBtn = document.querySelector<HTMLButtonElement>(CombinationMap.bulkCombinationFormBtn);
+    const bulkCombinationsBtn = this.tabContainer.querySelector(CombinationMap.bulkCombinationFormBtn) as HTMLButtonElement;
+
+    // @todo: This is hard-coded but when the other bulk actions will be added this needs to be more generic via
+    //        the use of data attributes:
+    //           data-bulk-url: is the url to call for each selected IDs, by default only the ID is necessary in the data
+    //           data-form-url: indicates that a form must be opened, upon submit the data is used to be sent to the bulk-url
+    bulkFormBtn?.addEventListener('click', () => this.showFormModal(
+      bulkFormBtn.dataset.formUrl as string,
+      bulkCombinationsBtn.dataset.modalConfirmLabel as string,
+      bulkCombinationsBtn.dataset.modalCancelLabel as string,
+    ));
+  }
+
+  private showFormModal(formUrl: string, confirmButtonLabel: string, closeButtonLabel: string): void {
+    const selectedCombinationsCount = this.getSelectedCheckboxes().length;
+
+    let initialSerializedData: string;
+    const iframeModal = new IframeModal({
+      id: CombinationMap.bulkFormModalId,
+      iframeUrl: formUrl,
+      autoSizeContainer: 'form[name="bulk_combination"]',
+      closable: true,
+      confirmButtonLabel: confirmButtonLabel.replace(/%combinations_number%/, String(selectedCombinationsCount)),
+      closeButtonLabel,
+      onLoaded: (iframe: HTMLIFrameElement) => {
+        // Disable submit button as long as the form data has not changed
+        iframeModal.modal.confirmButton?.setAttribute('disabled', 'disabled');
+
+        if (iframe.contentWindow) {
+          // eslint-disable-next-line max-len
+          const form: HTMLFormElement | null = iframe.contentWindow.document.querySelector<HTMLFormElement>('form[name="bulk_combination"]');
+
+          if (form) {
+            initialSerializedData = this.serializeForm(form);
+            form.addEventListener('change', () => {
+              const currentSerializedData: string = this.serializeForm(form);
+
+              if (currentSerializedData === initialSerializedData) {
+                iframeModal.modal.confirmButton?.setAttribute('disabled', 'disabled');
+              } else {
+                iframeModal.modal.confirmButton?.removeAttribute('disabled');
+              }
+            });
+          }
+        }
+      },
+      confirmCallback: () => {
+        if (iframeModal.modal.iframe.contentWindow) {
+          // eslint-disable-next-line max-len
+          const form: HTMLFormElement | null = iframeModal.modal.iframe.contentWindow.document.querySelector<HTMLFormElement>('form[name="bulk_combination"]');
```

Comment:
that's a long line
😄

Reply:
I guess it could be fixed by destructuring

Reply:
Indeed, I'm gonna extract this in a method